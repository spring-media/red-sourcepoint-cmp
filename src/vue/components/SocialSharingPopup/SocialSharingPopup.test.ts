import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { SocialSharingPopup } from './';

const loadPrivacyManagerModal = jest.fn();

const privacyManagerStub = Vue.extend({
  name: 'PrivacyManager',
  render() {
    if (this.$scopedSlots.default) {
      return this.$scopedSlots.default({
        loadPrivacyManagerModal,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }) as any;
    }
  },
});

describe('SocialSharingPopup', () => {
  afterEach(() => {
    loadPrivacyManagerModal.mockReset();
  });

  it('should initially render visible without any errors', () => {
    expect(mount(SocialSharingPopup, { propsData: { privacyManagerId: 12345 } }).element).toMatchSnapshot();
  });

  it('should be not visible if initialVisibility is set to false', function () {
    const wrapper = mount(SocialSharingPopup, {
      propsData: { privacyManagerId: 12345, initialVisibility: false },
    });

    expect(wrapper.isVisible()).toBe(false);
  });

  it('should call the toggleVisibility method on close button', function () {
    const wrapper = mount(SocialSharingPopup, {
      propsData: { privacyManagerId: 12345, initialVisibility: true },
    });

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (wrapper.vm as any).toggleVisibility = jest.fn();
    wrapper.find('.social-sharing-popup__button--close').trigger('click');
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    expect((wrapper.vm as any).toggleVisibility).toHaveBeenCalledTimes(1);
  });

  it('should open the privacy manager', () => {
    const wrapper = mount(SocialSharingPopup, {
      stubs: { PrivacyManager: privacyManagerStub },
      propsData: { privacyManagerId: 12345, initialVisibility: true },
    });

    wrapper.find('.social-sharing-popup__button--accept').trigger('click');

    expect(loadPrivacyManagerModal).toHaveBeenCalledWith(12345);
    expect(loadPrivacyManagerModal).toHaveBeenCalledTimes(1);
  });
});
