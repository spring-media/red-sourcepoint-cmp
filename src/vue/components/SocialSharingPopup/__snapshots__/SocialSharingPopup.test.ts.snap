// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialSharingPopup should initially render without any errors 1`] = `
<div
  class="social-sharing-popup__container"
>
  <div
    class="social-sharing-popup__headline"
  >
    
        Deine Datensicherheit bei der Nutzung der Teilen-Funktion
      
  </div>
   
  <div
    class="social-sharing-popup__description"
  >
    
        Um diesen Artikel oder andere Inhalte über soziale Netzwerke zu teilen, brauchen wir deine Zustimmung für
        
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      
          diesen Zweck der Datenverarbeitung
        
    </a>
  </div>
   
  <div
    class="social-sharing-popup__button-container"
  >
    <button
      class="social-sharing-popup__button social-sharing-popup__button--close"
    >
      
          Schließen
        
    </button>
     
    <button
      class="social-sharing-popup__button social-sharing-popup__button--accept"
    >
      
            Akzeptieren
          
    </button>
  </div>
</div>
`;
